src/Umbraco.Core/Deploy/ArtifactBase.cs

```diff
@@ -33,7 +33,7 @@ protected ArtifactBase(TUdi udi, IEnumerable<ArtifactDependency>? dependencies =
     /// <inheritdoc />
     public IEnumerable<ArtifactDependency> Dependencies
     {
-        get => _dependencies;
+        get => _dependencies ??= Array.Empty<ArtifactDependency>();
         set => _dependencies = value.OrderBy(x => x.Udi);
```

Comment:
We might also want to evaluate the `OrderBy` when setting the value, so this isn't done each time the collection is iterated (although that will take the performance hit beforehand):
```suggestion
        set => _dependencies = value.OrderBy(x => x.Udi).ToArray();
```

Reply:
Usually we are only talking about a handful of dependencies, so likely not a huge deal either way.  But agree it probably makes sense to do it once upfront rather than potentially several times.